File: backend/api/src/handler/handler.py
from flask import Blueprint
from json import dumps
from datetime import datetime

errors_bp = Blueprint("errors", __name__)


class HandlerException(Exception):
    def __init__(self, message, status=None, payload=None):
        super().__init__()
        self.message = message
        if status is not None:
            self.status = status
        self.timestamp = str(datetime.now())
        self.payload = payload


@errors_bp.app_errorhandler(HandlerException)
def handler(exc):
    return dumps(exc.__dict__), exc.status

File: backend/api/src/handler/handler.go
package handler

import (
	"encoding/json"
	"net/http"
	"time"
)

type defaultError struct {
	StatusCode int       `json:"statusCode"`
	Message    string    `json:"mensagem"`
	TimeError  time.Time `json:"timestamp"`
	Path       string    `json:"path"`
}

// Handler - Handler function
func Handler(w http.ResponseWriter, r *http.Request, status int, message string) {
	de := defaultError{
		Message:    message,
		StatusCode: status,
		TimeError:  time.Now(),
		Path:       r.RequestURI,
	}
	w.WriteHeader(de.StatusCode)
	json.NewEncoder(w).Encode(&de)
}
